auth/src/main/java/com/firebase/ui/auth/util/smartlock/SignInDelegate.java

```diff
@@ -0,0 +1,427 @@
+    }
+
+    private void redirectToIdpSignIn(String email, String accountType) {
+        Intent nextIntent;
+        switch (accountType) {
+            case IdentityProviders.GOOGLE:
+                nextIntent = IdpSignInContainerActivity.createIntent(
+                        getContext(),
+                        mHelper.getFlowParams(),
+                        GoogleAuthProvider.PROVIDER_ID,
+                        email);
+                break;
+            case IdentityProviders.FACEBOOK:
+                nextIntent = IdpSignInContainerActivity.createIntent(
+                        getContext(),
+                        mHelper.getFlowParams(),
+                        FacebookAuthProvider.PROVIDER_ID,
+                        email);
+                break;
+            case IdentityProviders.TWITTER:
+                nextIntent = IdpSignInContainerActivity.createIntent(
+                        getContext(),
+                        mHelper.getFlowParams(),
+                        TwitterAuthProvider.PROVIDER_ID,
+                        email);
+                break;
+            default:
+                Log.w(TAG, "unknown provider: " + accountType);
+                nextIntent = AuthMethodPickerActivity.createIntent(
+                        getContext(),
+                        mHelper.getFlowParams());
+        }
+        startActivityForResult(nextIntent, RC_IDP_SIGNIN);
+    }
+
+    private void finish(int resultCode, Intent data) {
+        if (mGoogleApiClient != null) {
+            mGoogleApiClient.disconnect();
+        }
+        mHelper.dismissDialog();
+
+        try {
+            if (resultCode == RESULT_OK) {
+                ((AuthUI.SignInResult) getActivity()).onAuthUISuccess(data);
+            } else {
+                ((AuthUI.SignInResult) getActivity()).onAuthUIFailure(resultCode);
+            }
+        } catch (ClassCastException e) {
+            Log.e(TAG, getActivity().toString()
+                    + " must implement AuthUI.AuthUIResult to receive sign in results");
+        }
+
+        getActivity().getSupportFragmentManager().beginTransaction().remove(this).commit();
+    }
+
+    /**
+     * Check if there is an active or soon-to-be-active network connection.
+     */
+    private boolean hasNetworkConnection() {
+        ConnectivityManager manager =
+                (ConnectivityManager) getActivity().getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        return manager != null
+                && manager.getActiveNetworkInfo() != null
+                && manager.getActiveNetworkInfo().isConnectedOrConnecting();
+    }
+
+    public static void delegateSignIn(FragmentActivity activity, FlowParameters parameters) {
+        FragmentManager fm = activity.getSupportFragmentManager();
+        FragmentTransaction ft = fm.beginTransaction();
+
+        Fragment fragment = fm.findFragmentByTag(TAG);
+        if (fragment == null || !(fragment instanceof SignInDelegate)) {
```

Comment:
What about the case where the Fragment is not null?  Is that handled by the fact that this is retained?